### PR TITLE
feat: send in Zaak close

### PR DESCRIPTION
**Pull request by hsiliev**
Send mail on Zaak close without checking for policy

Solves: PZ-2901

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/mail/model/MailAdres.kt [hunk 1]
```diff
@@ -7,11 +7,16 @@ package net.atos.zac.mail.model
 import jakarta.json.bind.annotation.JsonbProperty
 import jakarta.mail.Address
 import jakarta.mail.internet.InternetAddress
+import net.atos.zac.util.ValidationUtil
 
 class MailAdres(
     @field:JsonbProperty("Email") var email: String,
     @field:JsonbProperty("Name") var name: String?
 ) {
+    init {
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/mail/model/MailAdres.kt:16]: The @Valid annotation does not work here?

**Review by edgarvonk — COMMENTED** [review]
looks good!

File: src/main/kotlin/net/atos/zac/mail/model/MailAdres.kt (hunk 1 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/mail/model/MailAdres.kt:16]: We construct this in converters so no jsonb validation takes place
Additionally `@Email` is not very strict and allows emails like `a@b`, while we want `a@b.domain`

File: src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt [hunk 2]
```diff
@@ -217,4 +230,38 @@ class PlanItemsRESTServiceTest : BehaviorSpec({
...
+
+        every { zrcClientService.readZaak(zaak.uuid) } returns zaak
+        every { policyService.readZaakRechten(zaak) } returns createZaakRechtenAllDeny(startenTaak = true)
+        every { policyService.checkZaakAfsluitbaar(zaak) } just runs
+        every { brcClientService.listBesluiten(zaak) } returns Optional.of(listOf(Besluit()))
+        every { zrcClientService.readResultaat(zaak.resultaat) } returns resultaat
+        every { zrcClientService.updateResultaat(any<Resultaat>()) } returns resultaat
+        every { mailService.sendMail(mailGegevens, any<Bronnen>()) } returns mailGegevens.body
+
+        When("Zaak is requested to close and send mail") {
```
Comment (edgarvonk) [inline src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt:248]:
```suggestion
        When("A user event to settle the zaak and send a corresponding email is planned") {
```

? bit hard to describe it but something like that?

File: src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt [hunk 3]
```diff
@@ -217,4 +230,38 @@ class PlanItemsRESTServiceTest : BehaviorSpec({
...
+                zaakUuid = zaak.uuid,
+                actie = UserEventListenerActie.ZAAK_AFHANDELEN,
+                restMailGegevens = restMailGegevens
+            )
+            every { restMailGegevensConverter.convert(restMailGegevens) } returns mailGegevens
+            every { cmmnService.startUserEventListenerPlanItem(restUserEventListenerData.planItemInstanceId) } just runs
+
+            planItemsRESTService.doUserEventListenerPlanItem(restUserEventListenerData)
+
+            Then("it sends mail") {
```
Comment (edgarvonk) [inline src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt:260]:
```suggestion
            Then("the zaak is settled and the email is sent") {
```

**Review by edgarvonk — CHANGES_REQUESTED** [review]
nice!

File: src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt (hunk 2 above)
Comment (hsiliev) [reply src/test/kotlin/net/atos/zac/app/planitems/PlanItemsRESTServiceTest.kt]: Better :) Thanks!

Committed your suggestion

**Review by edgarvonk — APPROVED** [review]

**Pull request merged**